model/src/main/kotlin/DependencyNavigator.kt

```diff
@@ -215,7 +215,7 @@ interface DependencyNavigator {
 
             val newParents = item.parents + item.pkgRef.id
             item.pkgRef.visitDependencies { dependencyNodes ->
-                dependencyNodes.forEach { node -> queue.offer(QueueItem(node, newParents)) }
+                dependencyNodes.forEach { node -> queue.offer(QueueItem(node.getStableReference(), newParents)) }
```

Comment:
Just for my information, how would you generally identify such code spots where a call to `getStableReference()` is missing? Apparently, before this commit no test was broken, so we do not seem to have test coverage for this, or?

Reply:
Before this commit, the only `getStableReference()` implementation existing was for `PackageReference`, which returned itself. With `DependencyGraphNavigator`, there is now a new implementation, and this one would break tests.

That said, I am not very happy myself with this `getStableReference()` function and the necessity to call it explicitly; this is quite easy to forget. The main reason to introduce it was because of the `RuleSet` class, which keeps track of the `DependencyNode`s that are ancestors of the current package passed to a dependency rule. Maybe, in a later refinement, it is possible to provide an implementation of the _Visitor_ pattern or something like that, that allows a controlled iteration over all dependencies and hides these details.